src/ws/ws_manager.rs

```diff
@@ -32,6 +30,7 @@ use ethers::types::H160;
 struct SubscriptionData {
     sending_channel: UnboundedSender<Message>,
     subscription_id: u32,
+    identifier : String,
```

Comment:
nit: no space between member variable name and colon, i.e. `identifier: String`, also just call it `id` for brevity

Reply:
identifier is consistent with naming in existing subscribe / unsubscribe methods so thats why I used it originally, but no view from me and changed to id for consistency in the struct.